### DISCUSSION TITLE
Which pins for M7/M9 GRBL for esp32S?

### DISCUSSION THREAD

**Discussion by hfsdov** [grblHAL/core#573] [General]
My THC uses m7 and m9 as part of its anti dive configuration. It needs the ability to set 5v high and low based on sheetcam rules. 

Which pins support these commands. I looked thru the code and it looks like pin 21. Is that correct?

**Comment by terjeio** [comment]
Most pins can be used for coolant output, which one is is determined by the board map. The `$pins` command outputs the mapping and is the easiest way to find out. Alternatively consult the board map file used for the build.
